experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,256 @@
+# I would like to credit Dario Mathiä who produced an initial version of the following code
+
+import Chevie: complex_reflection_group, charinfo, CharTable, CyclotomicNumbers, Cyc
+export wreath_macs
+
+# Tools
+
+#gives indices of a subarray
+function subarray_indices(sub,arr)
+  res = Int[]
+  for i in 1:length(sub)
+    push!(res,findfirst(x->x==sub[i],arr))
+  end
+  return res
+end
+
+# Computes the b-invariant of a partition
+function b1(lambda::Vector{Int64})
+  if length(lambda)==0
+    return 0
+  end
+  return sum(i->(i-1)*lambda[i],1:length(lambda))
+end
+
+# Computes the b-invariant of a multipartition
+function btot(lbb::Multipartition)
+  return b1(map(lambda -> sum(lambda)[1], lbb)) + length(lbb)*sum(i -> b1(data(lbb[i])),1:length(lbb))
+end
+
+function beta_number_to_partition(beta::Vector{Integer})
+  lb=Integer[]
+  lbeta=length(beta)
+  for j in 1:lbeta
+    nbholes=beta[j]-beta[1]-(j-1)
+    if nbholes >= 1
+    pushfirst!(lb, nbholes)
+    end
+  end
+  return partition(lb)
+end
+
+# Residue notation is in terms of rows and columns
+
+function core(coroot::Vector{Int64},r::Integer)
+  beta=Integer[]
+  m=minimum(coroot)
+  M=maximum(coroot)
+  for k in m:M
+    for j in 1:r
+       if k <= coroot[j]
+          append!(beta, k*r+(j-1))
```

Comment:
```suggestion
          push!(beta, k*r+(j-1))
```
`push!` adds one entry to a vector, while `append!` appends multiple things. here both work, but `push!` is more efficient